wagtail_streamfield_migration_toolkit/migrate_operation.py

```diff
@@ -148,12 +151,20 @@ def migrate_stream_data_forward(self, apps, schema_editor):
 
             altered_raw_data = json.loads(revision.content[self.field_name])
             for operation, block_path_str in self.operations_and_block_paths:
-                altered_raw_data = utils.apply_changes_to_raw_data(
-                    raw_data=altered_raw_data,
-                    block_path_str=block_path_str,
-                    operation=operation,
-                    streamfield=getattr(model, self.field_name),
-                )
+                try:
+                    altered_raw_data = utils.apply_changes_to_raw_data(
+                        raw_data=altered_raw_data,
+                        block_path_str=block_path_str,
+                        operation=operation,
+                        streamfield=getattr(model, self.field_name),
+                    )
+                except utils.InvalidBlockDefError as e:
+                    # TODO this check might be a problem with wagtail 3.0
+                    if revision.id not in live_and_latest_revision_ids:
+                        logger.exception(e)
+                        continue
+                    else:
+                        raise e
```

Comment:
I think some additional context would be useful for the user, for example:
- is it the instance or a revision that caused the error?
- if it's an instance, what's the model and primary key?
- if it's a revision, what's its ID / contenttype / timestamp?

Reply:
I agree, this would be really useful.